compiler/luci-interpreter/src/kernels/CMakeLists.txt

```diff
@@ -1,234 +1,60 @@
-target_link_libraries(luci_interpreter_kernels
-    PUBLIC luci_interpreter_core
-    PRIVATE nncc_common Threads::Threads)
+        BinaryOpCommon.h
+        Utils.h
+        Utils.cpp
+        ${TensorFlowSource_DIR}/tensorflow/lite/kernels/internal/quantization_util.cc)
+
+macro(REGISTER_KERNEL NODE)
+  list(APPEND SOURCES "${NODE}.h")
+  list(APPEND SOURCES "${NODE}.cpp")
+endmacro(REGISTER_KERNEL)
+
+if(EXISTS "${KERNEL_REGISTER_FILE}")
+  include(${KERNEL_REGISTER_FILE})
+endif(EXISTS "${KERNEL_REGISTER_FILE}")
+
+add_library(${LUCI_INTERPRETER_KERNELS} STATIC ${SOURCES})
+set_target_properties(${LUCI_INTERPRETER_KERNELS} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+target_include_directories(${LUCI_INTERPRETER_KERNELS} PUBLIC ${LUCI_INTERPRETER_SOURCE_DIR})
+target_include_directories(${LUCI_INTERPRETER_KERNELS} SYSTEM PRIVATE
+        "${TensorFlowRuySource_DIR}"
+        "${TensorFlowGEMMLowpSource_DIR}"
+        "${TensorFlowEigenSource_DIR}"
+        "${TensorFlowSource_DIR}")
+
+if (NOT LUCI_MICRO)
+  target_link_libraries(${LUCI_INTERPRETER_KERNELS}
+          PUBLIC ${LUCI_INTERPRETER_CORE}
+          PRIVATE nncc_common Threads::Threads)
+else ()
+  target_link_libraries(${LUCI_INTERPRETER_KERNELS}
+          PUBLIC ${LUCI_INTERPRETER_CORE})
+endif()
 
 if(NOT ENABLE_TEST)
   return()
 endif(NOT ENABLE_TEST)
 
 nnas_find_package(GTest REQUIRED)
 
-set(TEST_SOURCES
-    Add.test.cpp
-    ArgMax.test.cpp
-    AveragePool2D.test.cpp
-    BatchToSpaceND.test.cpp
-    Cast.test.cpp
-    Concatenation.test.cpp
-    Conv2D.test.cpp
-    DepthToSpace.test.cpp
-    DepthwiseConv2D.test.cpp
-    Div.test.cpp
-    Elu.test.cpp
-    Exp.test.cpp
-    Floor.test.cpp
-    FloorDiv.test.cpp
-    Equal.test.cpp
-    FullyConnected.test.cpp
-    Greater.test.cpp
-    GreaterEqual.test.cpp
-    If.test.cpp
-    InstanceNorm.test.cpp
-    L2Normalize.test.cpp
-    L2Pool2D.test.cpp
-    LeakyRelu.test.cpp
-    Less.test.cpp
-    LessEqual.test.cpp
-    LocalResponseNormalization.test.cpp
-    LogicalAnd.test.cpp
-    LogicalNot.test.cpp
-    LogicalOr.test.cpp
-    Logistic.test.cpp
-    LogSoftmax.test.cpp
-    Maximum.test.cpp
-    MaxPool2D.test.cpp
-    Mean.test.cpp
-    Minimum.test.cpp
-    Mul.test.cpp
-    Neg.test.cpp
-    NotEqual.test.cpp
-    Pack.test.cpp
-    Pad.test.cpp
-    PadV2.test.cpp
-    Pow.test.cpp
-    PRelu.test.cpp
-    Relu.test.cpp
-    Relu6.test.cpp
-    Reshape.test.cpp
-    ResizeBilinear.test.cpp
-    ResizeNearestNeighbor.test.cpp
-    ReverseV2.test.cpp
-    Rsqrt.test.cpp
-    Slice.test.cpp
-    Softmax.test.cpp
-    SpaceToBatchND.test.cpp
-    SpaceToDepth.test.cpp
-    Split.test.cpp
-    StridedSlice.test.cpp
-    Sqrt.test.cpp
-    Square.test.cpp
-    SquaredDifference.test.cpp
-    Squeeze.test.cpp
-    Sub.test.cpp
-    Tanh.test.cpp
-    Transpose.test.cpp
-    TransposeConv.test.cpp
-    Unpack.test.cpp
-    While.test.cpp)
+macro(REGISTER_KERNEL NODE)
+  if(EXISTS "${NODE}.test.cpp")
```

Comment:
If author mistakes in name of test file it will not be compiled, maybe remove this `IF`?

Reply:
There is no test for some kernels e.g. MirrorPad.test.cpp, I suggest to add an empty file with TODO